src/validate_utf8.ts

```diff
@@ -16,32 +29,17 @@ const CONTINUING_CHAR = 0x80;
  * @param end - The index to end validating
  */
 export function validateUtf8(
-  bytes: { [index: number]: number },
+  buffer: Uint8Array,
   start: number,
-  end: number
-): boolean {
-  let continuation = 0;
-
-  for (let i = start; i < end; i += 1) {
-    const byte = bytes[i];
-
-    if (continuation) {
-      if ((byte & FIRST_TWO_BITS) !== CONTINUING_CHAR) {
-        return false;
-      }
-      continuation -= 1;
-    } else if (byte & FIRST_BIT) {
-      if ((byte & FIRST_THREE_BITS) === TWO_BIT_CHAR) {
-        continuation = 1;
-      } else if ((byte & FIRST_FOUR_BITS) === THREE_BIT_CHAR) {
-        continuation = 2;
-      } else if ((byte & FIRST_FIVE_BITS) === FOUR_BIT_CHAR) {
-        continuation = 3;
-      } else {
-        return false;
-      }
+  end: number,
+  fatal: boolean
+): string {
+  if (fatal) {
+    try {
+      return new TextDecoder('utf8', { fatal }).decode(buffer.slice(start, end));
```

Comment:
the text decoder could be lazy loaded

Reply:
By 'lazy loading,' I'm thinking you mean store the decoder instances so they are created only once at the first invocation of the `validateUtf8` method (see [commit 8a07891](https://github.com/mongodb/js-bson/pull/676/commits/8a07891c057a600d8fe3c0c2cb1dbbb5f5ed182d).

Reply:
If the `TextDecoder` is still slower than `Buffer.toString`, and the following are true:
- fatal/strict validation is on
- the string contains a replacement character
- the user is on a node environment

then this may have a negative performance impact when a user opts for validating utf8 with the driver.